### PR TITLE
JWT authentication instead of certificates

### PR DESCRIPTION
**Pull request by auvn**

### CODE REVIEW

File: apns/client_token.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package apns
+
+import (
+	"fmt"
+
+	"github.com/sideshow/apns2/token"
```
Comment (mkorolyov) [inline apns/client_token.go:6]: do we really need this dependency?

File: apns/transport.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+// RoundTripperJWTDecorator an implementation of http.RoundTripper interface
+// with ability to specify authorization JWT token for each request.
+type RoundTripperJWTDecorator struct {
+	Parent http.RoundTripper
+	Token  *Token
+}
+
+func (d *RoundTripperJWTDecorator) RoundTrip(r *http.Request) (*http.Response, error) {
+	r.Header.Set("Authorization", fmt.Sprintf("bearer %s", d.Token.Bearer()))
```
Comment (mkorolyov) [inline apns/transport.go:16]: ifaik token should be refreshed every <timeout>.

File: apns/client_token.go (hunk 1 above)
Comment (auvn) [reply apns/client_token.go:6]: we could copy the logic from this package, so the dependency will be removed, don't know what's better :)

File: apns/transport.go (hunk 2 above)
Comment (auvn) [reply apns/transport.go:16]: it should be refreshed every hour, the library guarantees refresh every 50 mins

**Pull request merged**